File: src/common/py/component/component_ids.py
from enum import StrEnum


class ComponentType(StrEnum):
    """
    All known component types.
    """

    INFRASTRUCTURE = "infra"
    WEB = "web"


class ComponentUnit(StrEnum):
    """
    All known component units.
    """

    # Infrastructure
    SERVER = "server"
    CONNECTOR = "connector"
    DOMO = "domo"

    # Web
    FRONTEND = "frontend"

File: src/common/web/component/ComponentIDs.ts
/**
 * All known component types.
 */
export const enum ComponentType {
    Infrastructure = "infra",
    Web = "web"
}

/**
 * All known component units.
 */
export const enum ComponentUnit {
    // Infrastructure
    Server = "server",
    Connector = "connector",
    Domo = "domo",

    // Web
    Frontend = "frontend"
}
